src/core/server/server.ts

```diff
@@ -155,9 +165,14 @@ export class Server {
       http: httpSetup,
     });
 
+    const coreTelemetrySetup = this.coreTelemetry.setup({
+      savedObjectsStartPromise: this.savedObjectsStartPromise,
+    });
```

Comment:
Here is the setup/start process:

1. CoreTelemetryService setup _(depends on SavedObjectsService start)_
2. SavedObjectsService setup _(depends on CoreTelemetryService setup)_
3. CoreUsageData setup _(depends on CoreTelemetryService setup)_
4. SavedObjectsService start
5. CoreUsageData start, immediately fetching usage data and querying for any persisted telemetry data in the process

The CoreTelemetryService needs a SavedObjectsRepository so it can persist telemetry data to Elasticsearch. I spoke with @pgayvallet and we determined the best way to handle this messy circular dependency is to pass a promise to the CoreTelemetryService which resolves after the SavedObjectsService is started.

Reply:
Done in 7ecb815, thanks!